subprojects/diagnostics/src/integTest/groovy/org/gradle/api/tasks/diagnostics/DependencyInsightReportTaskIntegrationTest.groovy

```diff
@@ -1441,4 +1442,36 @@ org:foo:$displayVersion -> $selected
         "prefer '[1.0, 1.4]'; reject '1.4'" | '[1.0, 1.4]'   | "1.4 has a critical bug"                        | '1.3'
     }
 
+    def "shows published dependency reason"() {
```

Comment:
This shows that a published reason ends up in the report, but a more fundamental test will be to show it's in the resolution result (see previous comment).